backend/executor.go

```diff
@@ -78,9 +78,11 @@ func WithStreamShutdownChannel(c <-chan any) grpcExecutorOptions {
 
 func NewGrpcExecutor(grpcServer *grpc.Server, be Backend, logger Logger, opts ...grpcExecutorOptions) Executor {
 	executor := &grpcExecutor{
-		workItemQueue: make(chan *protos.WorkItem),
-		backend:       be,
-		logger:        logger,
+		workItemQueue:        make(chan *protos.WorkItem),
+		backend:              be,
+		logger:               logger,
+		pendingOrchestrators: &sync.Map{},
+		pendingActivities:    &sync.Map{},
```

Comment:
Just wondering, if we should have UTs here.
These issues can be caught there directly.

Reply:
The problem wasn't related to initialization. Rather, it was because I was manipulating local stack copies of these maps instead of allocating them on the heap. A carefully constructed unit test might have caught this, but my preference is to instead add black-box integration test coverage, which are more comprehensive and easier to maintain.